aten/src/ATen/native/Convolution.cpp

```diff
@@ -207,6 +218,43 @@ static void check_input_shape_forward(const at::Tensor& input,
              "Given weight of size ", weight.sizes(),
              ", expected bias to be 1-dimensional with ", weight.size(0), " elements",
              ", but got bias of size ", bias.sizes(), " instead");
+
+    for (int i = 2; i < k; ++i) {
+      input_shape.push_back(input.size(i) + 2 * padding[i-2]);
+      kernel_shape.push_back(weight.size(i));
+      if (input_shape.back() < kernel_shape.back()) {
+        kernel_size_correct = false;
+      }
+
+      // TODO: Should output padding be considered while computing output_shape?
+      output_shape.push_back(div_rtn<int64_t>(input_shape.back() - kernel_shape.back(), stride[i-2]) + 1);
+      if (output_shape.back() < 0) {
+        output_size_correct = false;
+      }
+    }
+
+    AT_CHECK(input_shape.size() == kernel_shape.size() && input_shape.size() == output_shape.size(),
+             "Inconsistent shape between Input, Kernel and Output");
+
+    if (!kernel_size_correct || !output_size_correct) {
+      // If kernel size or output size is incorrect
+      std::ostringstream input_ss;
+      std::ostringstream kernel_ss;
+      std::ostringstream output_ss;
+      std::string separator = "";
+
+      for (int i = 0, len = input_shape.size(); i < len; ++i) {
+        input_ss << separator << input_shape[i];
+        kernel_ss << separator << kernel_shape[i];
+        output_ss << separator << output_shape[i];
+        separator = " x ";
+      }
+
+      AT_CHECK(kernel_size_correct, "Calculated padded input size per channel: (", input_ss.str(), "). "
+               "Kernel size: (", kernel_ss.str(), "). Kernel size can't be greater than actual input size");
+      AT_CHECK(output_size_correct, "Calculated padded input size per channel: (", input_ss.str(), "). "
+               "Calculated output size per channel: (", output_ss.str(), "). Output size is too small");
+    }
   } else { // transposed
     AT_CHECK(input.size(1) == weight.size(0),
```

Comment:
do we need to do similar checks for transposed as well?

Reply:
No. transposed convolution are not dependent on kernel size and input shape. output will determine on kernel size but not restricted on kernel size.